### PR TITLE
Rc/2.1.0 to main

### PR DESCRIPTION
**Pull request by mingstat**
Added `load_files()` to load data using explicit file paths. The code of the legacy `load_data()` function remains unchanged.

### CODE REVIEW

File: R/dvloader.R [hunk 1]
```diff
@@ -59,3 +59,62 @@ load_data <- function(sub_dir = NULL, file_names, use_wd = FALSE, prefer_sas = F
...
+
+  read_file_and_attach_metadata <- function(path) {
+    extension <- tools::file_ext(path)
+
+    if (toupper(extension) == "RDS") {
+      data <- readRDS(path)
+    } else if (toupper(extension) == "SAS7BDAT") {
+      data <- haven::read_sas(path)
+    } else {
+      stop("Internal error. Report this message to the maintainer of the `dv.loader` package.")
```
[the remark this answers is not in the document]
Comment (mingstat) [reply R/dvloader.R]: As suggested, the error message has been revised. https://github.com/Boehringer-Ingelheim/dv.loader/commit/a179d2bf75705a07fabd93f49a8bbab3834ee359

File: README.md [hunk 2]
```diff
@@ -1,30 +1,75 @@
...
-The package provides a simple function, `load_data()`, which loads R and SAS data files into memory. 
-Loading data from SQL databases is not yet supported. The function returns a list named by the file names passed, 
-and containing data frames, along with metadata for that table. By default, the function will look for files in a
-sub-directory `sub_dir` of the base path defined by a environment variable "RXD_DATA". You can check if the base path
-is set by running `Sys.getenv("RXD_DATA")`. A single file or multiple files can be loaded at once. 
-To make the loading process faster for large datasets, it is suggested that '.sas7bdat' files are converted to 
-'.RDS' files. The function will prefer '.RDS' files over '.sas7bdat' files by default.
+The `dv.loader` package provides functionality for loading `.rds` and `.sas7bdat` data files into R. It offers two main functions:
+
+- `load_data()`: A legacy function that loads data files from a specified subdirectory of the base path defined by the environment variable "RXD_DATA".
```
[the remark this answers is not in the document]
Comment (mingstat) [reply README.md]: This has been fixed by updating README and vignettes. https://github.com/Boehringer-Ingelheim/dv.loader/pull/8/commits/61f614ed1cb085eb5bb9ee06826283641f4be7b6 https://github.com/Boehringer-Ingelheim/dv.loader/pull/8/commits/3aa99b893321eedb79a01b0f11867db4e6a7a393

File: README.md [hunk 3]
```diff
@@ -1,30 +1,75 @@
...
+- `load_files()`: A newer function that provides more flexibility by accepting file paths directly and supports custom names for the returned data list.
+
+NOTE: The `load_files()` function is recommended for its enhanced capabilities and flexibility.
```
[the remark this answers is not in the document]
Comment (mingstat) [reply README.md]: The documentation now describes load_data() and load_files() as the two main functions for loading data, each designed for different use cases.

File: R/dvloader.R [hunk 4]
```diff
@@ -59,3 +59,62 @@ load_data <- function(sub_dir = NULL, file_names, use_wd = FALSE, prefer_sas = F
...
+#' @return [list] A named list of data frames, where each name is either:
+#'  - the name associated to the element in the `file_paths` argument, or, if not provided...
+#'  - the name of the file itself, after stripping it of its leading path and trailing extension
+#'
+#' @export
+load_files <- function(file_paths) {
+  checkmate::assert_character(file_paths, min.len = 1)
+  checkmate::assert_file_exists(file_paths, access = "r", extension = c("rds", "sas7bdat"))
+
+  read_file_and_attach_metadata <- function(path) {
```
Comment (mattkorb) [inline R/dvloader.R:79]: Should we put this function into utils.R?

Comment (mingstat) [reply R/dvloader.R]: This function has been moved to utils.R, allowing it to be used by both load_data() and load_files().

**Review by mattkorb — APPROVED** [review]

**Pull request merged**